Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoASA.yaml

```diff
@@ -0,0 +1,423 @@
+                | extend SimplifiedDeviceAction = trim(@"\s",SimplifiedDeviceAction)
+                | extend 
+                        EventStartTime = TimeGenerated,
+                        EventEndTime = TimeGenerated,
+                        EventVendor = "Cisco",
+                        EventProduct = "ASA",
+                        EventCount = coalesce(EventCount,toint(1)),
+                        EventType = "NetworkSession",
+                        EventSchema = "NetworkSession",
+                        EventSchemaVersion = "0.2.4",
+                        DvcAction = case(SimplifiedDeviceAction has_any ("Teardown","Built"), "",
+                                        SimplifiedDeviceAction),
+                        SrcInterfaceName = tolower(SrcInterfaceName),
+                        DstInterfaceName = tolower(SrcInterfaceName)
+                | extend 
+                        EventResult = case(isnotempty(EventResult), EventResult,
+                                            DvcAction == "Allow", "Success",
+                                            "Failure"),
+                        SrcUsernameType = case(isnotempty(SrcUsername) and SrcUsername has "@", "UPN",
+                                               isnotempty(SrcUsername) and SrcUsername !has "@" and SrcUsername has "\\", "Windows",
+                                               isnotempty(SrcUsername), "Simple",
+                                               ""),
+                        DstUsernameType = case(isnotempty(DstUsername) and DstUsername has "@", "UPN",
+                                               isnotempty(DstUsername) and DstUsername !has "@" and DstUsername has "\\", "Windows",
+                                               isnotempty(DstUsername), "Simple",
+                                               "")
+                | lookup ProtocolLookup on Protocol
+                | project-rename 
+                        EventProductVersion = AliasDeviceVersion,
+                        EventOriginalType = AliasDeviceEventClassID,
+                        EventOriginalSeverity = OriginalLogSeverity,
+                        DvcOriginalAction = DeviceAction,
+                        EventMessage = Message,
+                        Dvc = Computer
+                | extend
+                        EventSeverity = iff(isempty(EventResult) or EventResult == "Success", "Informational", "Low"),
+                        NetworkDirection = case(isnotempty(CommunicationDirection), CommunicationDirection,
+                                    (SrcInterfaceName in (internalInterface) and DstInterfaceName in (externalInterface)) or DstInterfaceName in (externalInterface), "Outbound",
```

Comment:
I may be missing something, but this logical expression seems to be superfluous. If _DstInterfaceName in (externalInterface)_ is true, _SrcInterfaceName in (internalInterface)_ would not matter. Why do we need the left side of the _or_ operator? Same for the next line.